R/makeTreeSummarizedExperimentFromqiime2.R

```diff
@@ -1,93 +1,99 @@
-#' taxaqza_file <- system.file("extdata", "taxonomy.qza", package = "mia")
-#' sample_file <- system.file("extdata", "sample-metadata.tsv", package = "mia")
-#' treeqza_file <- system.file("extdata", "tree.qza", package = "mia")
-#' refseqqza_file <- system.file("extdata", "refseq.qza", package = "mia")
+#' featureTableFile <- system.file("extdata", "table.qza", package = "mia")
+#' taxonomyTableFile <- system.file("extdata", "taxonomy.qza", package = "mia")
+#' sampleMetaFile <- system.file("extdata", "sample-metadata.tsv", package = "mia")
+#' phyTreeFile <- system.file("extdata", "tree.qza", package = "mia")
+#' repSeqFile <- system.file("extdata", "refseq.qza", package = "mia")
 #' tse <- makeTreeSummarizedExperimentFromqiime2(
-#'   otu_qza = otuqza_file, taxa_qza = taxaqza_file,
-#'   sam_tab = sample_file, refseq_qza = refseqqza_file,
-#'   tree_qza = treeqza_file
+#'   featureTableFile = featureTableFile,
+#'   taxonomyTableFile = taxonomyTableFile,
+#'   sampleMetaFile = sampleMetaFile,
+#'   repSeqFile = repSeqFile,
+#'   phyTreeFile = phyTreeFile
 #' )
 #'
 #' tse
-makeTreeSummarizedExperimentFromqiime2 <- function(otu_qza,
-                                                   taxa_qza = NULL,
-                                                   sam_tab = NULL,
-                                                   refseq_qza = NULL,
-                                                   tree_qza = NULL) {
-    feature_tab <- read_qza(otu_qza)
-
-    if (!is.null(taxa_qza)) {
-        taxa_tab <- read_qza(taxa_qza)
-        taxa_tab <- subset_taxa_in_feature(taxa_tab, feature_tab)
-        taxa_tab <- parse_q2taxonomy(taxa_tab)
+makeTreeSummarizedExperimentFromqiime2 <- function(featureTableFile,
+                                                   taxonomyTableFile = NULL,
+                                                   sampleMetaFile = NULL,
+                                                   featureNamesAsRepseq = TRUE,
+                                                   repSeqFile = NULL,
+                                                   phyTreeFile = NULL) {
+    feature_tab <- .read_qza(featureTableFile)
+
+    if (!is.null(taxonomyTableFile)) {
+        taxa_tab <- .read_qza(taxonomyTableFile)
+        taxa_tab <- .subset_taxa_in_feature(taxa_tab, feature_tab)
+        taxa_tab <- .parse_q2taxonomy(taxa_tab)
     } else {
-        taxa_tab <- NULL
+        taxa_tab <- S4Vectors:::make_zero_col_DataFrame(nrow(feature_tab))
     }
 
-    if (!is.null(sam_tab)) {
-        sam_tab <- read_q2sample_meta(sam_tab)
+    if (!is.null(sampleMetaFile)) {
+        sample_meta <- .read_q2sample_meta(sampleMetaFile)
     } else {
-        sam_tab <- NULL
+        sample_meta <- S4Vectors:::make_zero_col_DataFrame(ncol(feature_tab))
     }
 
-    if (!is.null(tree_qza)) {
-        tree <- read_qza(tree_qza)
+    if (!is.null(phyTreeFile)) {
+        tree <- .read_qza(phyTreeFile)
     } else {
         tree <- NULL
     }
```

Comment:
If `tree` is not not `NULL`, it is `NULL` isn't it?

Reply:
if `phyTreeFile` is not not `NULL`,  `tree <- NULL`